diff --git a/src/path.rs b/src/path.rs
index 7ccc18102..48d181f66 100644
--- a/src/path.rs
+++ b/src/path.rs
@@ -561,7 +561,7 @@ fn path(&self) -> Option<WString> {
 /// set, use that directory; otherwise use the path `non_xdg_homepath` rooted in $HOME. Return the
 /// result; see the base_directory_t fields.
 #[cfg_attr(test, allow(unused_variables), allow(unreachable_code))]
-fn make_base_directory(xdg_var: &wstr, non_xdg_homepath: &wstr) -> BaseDirectory {
+fn make_base_directory(preferred_var: &wstr, xdg_var: &wstr, non_xdg_homepath: &wstr) -> BaseDirectory {
     #[cfg(test)]
     // If running under `cargo test`, contain ourselves to the build directory and do not try to use
     // the actual $HOME or $XDG_XXX directories. This prevents the tests from failing and/or stops
@@ -591,14 +591,19 @@ fn make_base_directory(xdg_var: &wstr, non_xdg_homepath: &wstr) -> BaseDirectory
 
     let mut path = WString::new();
     let used_xdg;
-    if let Some(xdg_dir) = vars.getf_unless_empty(xdg_var, EnvMode::GLOBAL | EnvMode::EXPORT) {
-        path = xdg_dir.as_string() + L!("/fish");
-        used_xdg = true;
+    if let Some(preferred_dir) = vars.getf_unless_empty(preferred_var, EnvMode::GLOBAL | EnvMode::EXPORT) {
+        path = preferred_dir.as_string();
+        used_xdg = false;
     } else {
-        if let Some(home) = vars.getf_unless_empty(L!("HOME"), EnvMode::GLOBAL | EnvMode::EXPORT) {
-            path = home.as_string() + non_xdg_homepath;
+        if let Some(xdg_dir) = vars.getf_unless_empty(xdg_var, EnvMode::GLOBAL | EnvMode::EXPORT) {
+            path = xdg_dir.as_string() + L!("/fish");
+            used_xdg = true;
+        } else {
+            if let Some(home) = vars.getf_unless_empty(L!("HOME"), EnvMode::GLOBAL | EnvMode::EXPORT) {
+                path = home.as_string() + non_xdg_homepath;
+            }
+            used_xdg = false;
         }
-        used_xdg = false;
     }
 
     let mut remoteness = DirRemoteness::Unknown;
@@ -714,13 +719,13 @@ pub fn path_remoteness(path: &wstr) -> DirRemoteness {
 }
 
 static DATA_DIRECTORY: LazyLock<BaseDirectory> =
-    LazyLock::new(|| make_base_directory(L!("XDG_DATA_HOME"), L!("/.local/share/fish")));
+    LazyLock::new(|| make_base_directory(L!("FISH_DATA_DIR"), L!("XDG_DATA_HOME"), L!("/.local/share/fish")));
 
 static CACHE_DIRECTORY: LazyLock<BaseDirectory> =
-    LazyLock::new(|| make_base_directory(L!("XDG_CACHE_HOME"), L!("/.cache/fish")));
+    LazyLock::new(|| make_base_directory(L!("FISH_CACHE_DIR"), L!("XDG_CACHE_HOME"), L!("/.cache/fish")));
 
 static CONFIG_DIRECTORY: LazyLock<BaseDirectory> =
-    LazyLock::new(|| make_base_directory(L!("XDG_CONFIG_HOME"), L!("/.config/fish")));
+    LazyLock::new(|| make_base_directory(L!("FISH_CONFIG_DIR"), L!("XDG_CONFIG_HOME"), L!("/.config/fish")));
 
 /// Appends a path component, with a / if necessary.
 pub fn append_path_component(path: &mut WString, component: &wstr) {
