diff --git a/src/env_universal_common.rs b/src/env_universal_common.rs
index e0b6e4578..66b10ccbd 100644
--- a/src/env_universal_common.rs
+++ b/src/env_universal_common.rs
@@ -6,7 +6,7 @@
 use crate::env::{EnvVar, EnvVarFlags, VarTable};
 use crate::flog::{FLOG, FLOGF};
 use crate::fs::{lock_and_load, rewrite_via_temporary_file, PotentialUpdate};
-use crate::path::path_get_config;
+use crate::path::path_get_data;
 use crate::wchar::{decode_byte_from_char, prelude::*};
 use crate::wcstringutil::{join_strings, LineIterator};
 use crate::wutil::{file_id_for_file, file_id_for_path_narrow, wrealpath, FileId, INVALID_FILE_ID};
@@ -659,7 +659,7 @@ mod fish3_uvars {
 
 /// Return the default variable path, or an empty string on failure.
 fn default_vars_path_directory() -> Option<WString> {
-    path_get_config()
+    path_get_data()
 }
 
 /// Test if the message msg contains the command cmd.
diff --git a/tests/checks/symlinks-not-overwritten.fish b/tests/checks/symlinks-not-overwritten.fish
index 5b3446352..b4d760a9b 100644
--- a/tests/checks/symlinks-not-overwritten.fish
+++ b/tests/checks/symlinks-not-overwritten.fish
@@ -1,9 +1,9 @@
 # RUN: fish=%fish %fish %s
 
 # fish_variables
-set -l target_file $XDG_CONFIG_HOME/fish/target_fish_variables
-set -l fish_variables $XDG_CONFIG_HOME/fish/fish_variables
-set -l backup_file $XDG_CONFIG_HOME/fish/fish_variables_backup
+set -l target_file $XDG_DATA_HOME/fish/target_fish_variables
+set -l fish_variables $XDG_DATA_HOME/fish/fish_variables
+set -l backup_file $XDG_DATA_HOME/fish/fish_variables_backup
 
 echo >$target_file
 cp $target_file $backup_file
